src/js/game/LevelMVC/LevelModel.js

```diff
@@ -781,11 +781,19 @@ module.exports = class LevelModel {
         break;
     }
 
-    if (shouldPlace === true) {
-      var block = new LevelBlock(blockType);
+    if (entity === this.agent) {
+      let ground = this.groundPlane.getBlockAt(position);
+      if (ground.blockType === "water" || ground.blockType === "lava") {
```

Comment:
I propose we move as much of this logic into `Player` and `Agent` as we can; something like an `entity.canPlaceBlock` method that takes as arguments the blockType to be placed and the ground block it's being placed on top of. Then, `Player` can define the `wheat iff famland` logic and `Agent` can define the water/lava logic.

That'll be particularly nice, because we're going to want to expand the water/lava logic to make sure that only actual _blocks_ can be placed in water or lava, not things like redstone or wheat or rails.

Reply:
Excellent point with placing redstone in water. I haven't even looked at what would happen with Redstone on the ground plane. I bet it is a hideous mess. Will address this on Monday.

Reply:
Checked up on this this morning:
redstone Wire can now no longer be placed in water, but additionally, it cannot be placed below the Agent either. In actual minecraft "place below" will always fail if there is ground below you, even if the block (like wire or rails) doesn't take up the full index. There is no way to pace blocks within the index the Agent is in, so for consistency I have not allowed to place rails / redstone in the same slot.